### PR TITLE
Transfer token api

### PR DESCRIPTION
**Pull request by thebevrishot**
closes https://github.com/zcoinofficial/ztm/issues/85

### CODE REVIEW

File: src/Ztm.WebApi/Controllers/ControllerHelper.cs [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        {
+            if (callbackRepository == null)
+            {
+                throw new ArgumentNullException(nameof(callbackRepository));
+            }
+
+            this.callbackRepository = callbackRepository;
+        }
+
+        public async Task<Callback> TryAddCallbackAsync(ControllerBase controller, CancellationToken cancellationToken)
```
Comment (ultimaweapon) [inline src/Ztm.WebApi/Controllers/ControllerHelper.cs]: The name of method is quite confusing. I think `RegisterCallbackAsync` looks better.

File: src/Ztm.WebApi/Models/TransferingRequest.cs [hunk 2]
```diff
@@ -3,7 +3,7 @@
 
 namespace Ztm.WebApi.Models
 {
-    public class TransferingRequest
+    public class TransfersRequest
```
[the remark this answers is not in the document]
Comment (thebevrishot) [reply src/Ztm.WebApi/Models/TransferingRequest.cs]: The api path is `/transfers`, I believe this is correct.

File: src/Ztm.WebApi.Tests/Ztm.WebApi.Tests.csproj [hunk 3]
```diff
@@ -29,6 +29,7 @@
     <ProjectReference Include="..\Ztm.Data.Entity.Testing\Ztm.Data.Entity.Testing.csproj" />
     <ProjectReference Include="..\Ztm.Testing\Ztm.Testing.csproj" />
     <ProjectReference Include="..\Ztm.Zcoin.Watching\Ztm.Zcoin.Watching.csproj" />
+    <ProjectReference Include="..\Ztm.Zcoin.Rpc.Tests\Ztm.Zcoin.Rpc.Tests.csproj" />
```
[the remark this answers is not in the document]
Comment (thebevrishot) [reply src/Ztm.WebApi.Tests/Ztm.WebApi.Tests.csproj]: Removed

File: src/Ztm.Zcoin.Rpc.Tests/RPCExceptionTesting.cs [hunk 4]
```diff
@@ -0,0 +1,28 @@
+using System.IO;
+using System.Text;
+using NBitcoin.RPC;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Serialization;
+
+namespace Ztm.Zcoin.Rpc.Tests
+{
+    public static class RPCExceptionTesting
```
[the remark this answers is not in the document]
Comment (thebevrishot) [reply src/Ztm.Zcoin.Rpc.Tests/RPCExceptionTesting.cs]: This class is used by external.

File: src/Ztm.Zcoin.Rpc.Tests/RPCExceptionTesting.cs [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    {
+        public static RPCException BuildException(object response, RPCErrorCode code, string message)
```
[the remark this answers is not in the document]
Comment (thebevrishot) [reply src/Ztm.Zcoin.Rpc.Tests/RPCExceptionTesting.cs]: Make sense

File: src/Ztm.WebApi/Models/TransferingRequest.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (ultimaweapon) [reply src/Ztm.WebApi/Models/TransferingRequest.cs]: After discussion `/transfers` endpoint will act as resource endpoint. So we need to rename this to `TransferRequest`.

**Pull request merged**